### ISSUE TITLE
class definition followed by a package-private object fails to process due to bug in scalafmt

### ISSUE THREAD

**Issue by kwisatzook** [scalameta/scalafmt#1491]
- **Version**: 2.0.1
- **Configuration**:
```
version = "2.0.1"
style = default
maxColumn = 100
docstrings = JavaDoc
align = none
align.tokens = []
align.openParenCallSite = false
align.openParenDefnSite = false
continuationIndent.extendSite = 2 # Scalaguide calls for two spaces
importSelectors = singleLine
verticalMultilineAtDefinitionSite = true
optIn.breakChainOnFirstMethodDot = false
optIn.configStyleArguments = false
optIn.breaksInsideChains = true
newlines.neverBeforeJsNative = true
newlines.penalizeSingleSelectMultiArgList = false
rewrite {
  rules = [RedundantParens, SortModifiers]
  sortModifiers {
    order = ["implicit", "override", "private", "protected", "final", "sealed", "abstract", "lazy"]
  }
}
includeCurlyBraceInSelectChains = false
danglingParentheses = false
```

## Steps

Given code like this:
```scala
package somepackage

object Flurp {
  class Banana(foo: String)
  private[somepackage] object Dummy
}
```

When I run scalafmt like this:

```bash
sbt scalafmtAll
```

## Problem

```
[error] <path>/src/main/scala/somepackage/Flurp.scala:4: error: Unable to format file due to bug in scalafmt
[error]   class Banana(foo: String)
[error]                           ^: <path>/src/main/scala/somepackage/Flurp.scala
```

If you remove the [somepackage] it is processed fine. 
If you change Dummy to be a class (keeping the private[somepackage], it's processed fine.
If you reorder the definitions, it is processed fine.

## Expectation

Scalafmt should not have a problem processing a package-private object after a class definition.

**Comment by tanishiking** [comment]
Thank you so much for reporting!
Hmm, maybe related to https://github.com/scalameta/scalafmt/issues/1383
In order to deconstruct the problem, could you give us a minimal working configuration that reproduces the error? @kwisatzook

**Comment by poslegm** [comment]
Minimal config for reproduce:

```
version = "2.0.1"
verticalMultilineAtDefinitionSite = true
```

**Comment by tanishiking** [comment]
I just released scalafmt 2.1.0 that contains https://github.com/scalameta/scalafmt/pull/1513, check it out :) Thank you very much @poslegm

**Comment by tanishiking** [comment]
Let me close this issue for 2.1.0.
If you still have any problems on this in 2.1.0, feel free to re-open this :) Thank you so much for reporting! @kwisatzook
